src/sqlancer/postgres/PostgresProvider.java

```diff
@@ -218,24 +220,51 @@ protected TestOracle getTestOracle(PostgresGlobalState globalState) throws SQLEx
 
     @Override
     public Connection createDatabase(PostgresGlobalState globalState) throws SQLException {
-        String url = "jdbc:postgresql://localhost:5432/test";
+        String username = globalState.getOptions().getUserName();
+        String password = globalState.getOptions().getPassword();
+        String entryPath = "test";
+        String entryURL = globalState.getDmbsSpecificOptions().connectionURL;
+        // trim URL to exclude "jdbc:"
+        if (entryURL.startsWith("jdbc:")) {
+            entryURL = entryURL.substring(5);
+        }
+        try {
+            URI uri = new URI(entryURL);
+            String userInfoURI = uri.getUserInfo();
+            String pathURI = uri.getPath();
+            if (userInfoURI != null) {
+                // username and password specified in URL take precedence
+                String[] userInfo = userInfoURI.split(":", 2);
+                username = userInfo[0];
+                password = userInfo[1];
+                int userInfoIndex = entryURL.indexOf(userInfoURI);
+                entryURL = entryURL.substring(0, userInfoIndex)
+                        + entryURL.substring(userInfoIndex + userInfoURI.length() + 1);
```

Comment:
I think it would be more readable to concatenate `username` and `password`, rather than doing `entryURL.substring(userInfoIndex + userInfoURI.length() + 1)` (if that's correct). Efficiency is not really a concern here, and I think doing so is less error prone.

Reply:
Actually, I am removing username and password here instead. Otherwise, if username and password are specified both in the DriverManager call and in the url passed to the Driver Manager, the conflict causes errors. So the concatenation I am doing is [urlBeforeUserInfo]+[urlAfterUserInfo].

Reply:
Okay, thanks for the explanation! If you think that additional variable names would not convey additional information, we can also leave it like it is.